File: 9.palindrome-number.kt
/*
 * @lc app=leetcode id=9 lang=kotlin
 *
 * [9] Palindrome Number
 */

// @lc code=start
class Solution {
    fun isPalindrome(x: Int): Boolean {
        return isStringPalindrome(x.toString())
    }

    fun isStringPalindrome(x: String): Boolean
    {
        for(i in 0 .. x.length/2) {
            if(x.get(i) != x.get(x.length-i-1)) return false
        }

        return true
    }
}
// @lc code=end

File: go/9.palindrome-number.go
/*
 * @lc app=leetcode id=9 lang=golang
 *
 * [9] Palindrome Number
 */

// @lc code=start
func isPalindrome(x int) bool {
	if x < 0 || (x != 0 && x%10 == 0) {
		return false
	}
	res := 0
	for x > res {
		res = res*10 + x%10
		x /= 10
	}
	return x == res || x == res/10
}
// @lc code=end

